single_kernel_mongo/managers/mongo.py

```diff
@@ -274,18 +274,22 @@ def add_user(self, relation: Relation):
                 )
 
     def update_user(self, relation: Relation) -> None:
-        """Add the user for this relation."""
+        """Update the user for this relation."""
         data_interface = DatabaseProviderData(
             self.model,
             relation.name,
         )
 
         username = f"relation-{relation.id}"
         password = data_interface.fetch_my_relation_field(relation.id, "password")
+
         with MongoConnection(self.state.mongo_config) as mongo:
             has_user = mongo.user_exists(username)
-        if has_user:
```

Comment:
This was a funny bug, glad you found it

Reply:
This has been raised by Mateo !
He realised he was lacking the access to his user but it had the accesses to the default database.
But this is good that it's fixed :)